pkg/config/polling_manager.go

```diff
@@ -219,6 +234,35 @@ func (cm *PollingProjectConfigManager) GetConfig() (ProjectConfig, error) {
 	return cm.projectConfig, nil
 }
 
+func (cm *PollingProjectConfigManager) setConfig(projectConfig ProjectConfig) error {
```

Comment:
I would remove locking mechanism here and minimize the number of calls to locking.
of course cm.sendConfigUpdateNotification() will need to be called in SyncConfig then